frontend/src/views/NftDisplay.vue

```diff
@@ -175,6 +318,13 @@ export default Vue.extend({
           ? this.Characters.options.address
           : this.Shields.options.address;
     },
+    canPurchase(): boolean {
+      return (
+        this.nftType === 'weapon' ||
+        this.nftType === 'shield' ||
+        this.ownCharacters.length < 4
+      );
+    },
```

Comment:
No need to check nftType against 'weapon' and 'shield' when canPurchase is used only in characters-list template. The first 2 conditiosn can be removed.